### DISCUSSION TITLE
Resize handles not visible while using freeTransform() in React app

### DISCUSSION THREAD

**Discussion by jibinjoy8897** [clientIO/joint#2528] [Q&A] [answered]
### Introduction

Added the below code as shown in freeTransform documentation.

    if (cellView.model instanceof joint.dia.Link) return;

    var freeTransform = new joint.ui.FreeTransform({ cellView: cellView });
    freeTransform.render();

The resize can be done from all 8 handles of the element but the handle points are not visible.In documentation styles and js imports are done.We are using a React app where we imported as mentioned in documentation in the style.css
@import '~@clientio/rappid/rappid.css';

Please suggest solution to make the handles visible.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
- Are you able to reproduce the issue here: https://github.com/clientIO/joint-plus-tutorial-react/blob/main/src/App.tsx
- When you say that the handles are invisible, does it mean they exist in the DOM but they have for example transparent color? Or they are not in the DOM at all?
- Do other JointJS+ UI plugins work as expected? Is the styling applied correctly to other plugins?

**Reply by jibinjoy8897** [reply]
* Getting an error on npm i (**npm ERR! enoent ENOENT: no such file or directory, open 'D:\POC\joint-plus-tutorial-react\rappid.tgz**'),since the rappid.tgz file is missing in the project.Is that the issue you are referring to?
* Yes they exist in DOM.I am able to resize node with these handles,but not visible.
* I believe only this line is needed to add styles in React right? (**import '~@clientio/rappid/rappid.css';**)

**Reply by kumilingus** [reply]
> Getting an error on npm i (npm ERR! enoent ENOENT: no such file or directory, open 'D:\POC\joint-plus-tutorial-react\rappid.tgz'),since the rappid.tgz file is missing in the project.Is that the issue you are referring to?

Please see the `README.md` of the project. You must provide `rappid.tgz` file first.

>Yes they exist in DOM.I am able to resize node with these handles,but not visible.

Aren't you using a custom theme? e.g. `joint.setTheme('my-theme')`
Because if the position of the handles is correct then the `rappid.css` is correctly applied.

> I believe only this line is needed to add styles in React right? (import '~@clientio/rappid/rappid.css';)
Yes.

**Reply by jibinjoy8897** [reply]
No we are not using custom theme.I just added the rect which was in the git project you send.There also the resize handle is not visible.Any idea where i went wrong?

**Reply by kumilingus** [reply]
Just added this to line *45* of `App.tsx` and everything works as expected...
```ts
paper.on('element:pointerclick', (elementView) => {
  const freeTransform = new ui.FreeTransform({
    cellView: elementView,
  });
  freeTransform.render();
});
```

**Reply by jibinjoy8897** [reply]
yes,it is working in the github project when adding the code you mentioned.Not working in my React application.Anything i missed out?

**Reply by kumilingus** [reply]
I don't know. You'd have to provide us with some steps to reproduce.

**Reply by jibinjoy8897** [reply]
I have referred the database demo (https://www.jointjs.com/demos/database) to use in our application case.In that please add freeTransform and check if handles are visible.In my case it was not visible.

**Reply by kumilingus** [reply]
The *Database* example sets a custom theme.

```js
setTheme('my-theme');
```

**Reply by jibinjoy8897** [reply, marked as answer]
By removing this line.Solved my issue.Thanks.
